pkg/daemon/ceph/osd/volume.go

```diff
@@ -214,7 +215,7 @@ func (a *OsdAgent) configureCVDevices(context *clusterd.Context, devices *Device
 	return osds, err
 }
 
-func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *DeviceOsdMapping, lvBackedPV bool) (string, string, error) {
+func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *DeviceOsdMapping, lvBackedPV bool) (string, string, string, error) {
```

Comment:
Please enhance the test for `initializeBlockPVC`. You might need to rebase this branch as we just merged a PR that was adding more coverage.

Reply:
I will rebase the new change and test the functionality, and update the test result later ...

Reply:
Rebased the new change for test initializeBlockPVC function and verified the wal device for OSD on PVC, set it resolved.
---------------------
$ kubectl get pod -A
NAMESPACE     NAME                                               READY   STATUS      RESTARTS   AGE
default       cluster-init-jrt46                                 1/1     Running     0          19s
default       cockroachdb-0                                      0/1     Running     0          24s
kube-system   calico-kube-controllers-76d4774d89-dg6nx           1/1     Running     0          8m17s
kube-system   calico-node-ptvcl                                  1/1     Running     0          8m17s
kube-system   coredns-66bff467f8-7wkgf                           1/1     Running     0          8m40s
kube-system   coredns-66bff467f8-zdvtz                           1/1     Running     0          8m40s
kube-system   etcd-master                                        1/1     Running     0          8m48s
kube-system   kube-apiserver-master                              1/1     Running     0          8m48s
kube-system   kube-controller-manager-master                     1/1     Running     0          8m48s
kube-system   kube-proxy-d5xcg                                   1/1     Running     0          8m40s
kube-system   kube-scheduler-master                              1/1     Running     0          8m48s
rook-ceph     csi-cephfsplugin-provisioner-6bcb7cdd75-v7wnr      0/5     Pending     0          2m57s
rook-ceph     csi-cephfsplugin-provisioner-6bcb7cdd75-w5gvn      5/5     Running     0          2m57s
rook-ceph     csi-cephfsplugin-wm6h6                             3/3     Running     0          2m57s
rook-ceph     csi-rbdplugin-62txw                                3/3     Running     0          2m58s
rook-ceph     csi-rbdplugin-provisioner-5bcf44d764-dtlv7         0/6     Pending     0          2m58s
rook-ceph     csi-rbdplugin-provisioner-5bcf44d764-v2ntl         6/6     Running     0          2m58s
rook-ceph     rook-ceph-crashcollector-master-66dbf85569-b7x7p   1/1     Running     0          93s
rook-ceph     rook-ceph-mgr-a-5994fb68c5-v5628                   1/1     Running     0          114s
rook-ceph     rook-ceph-mon-a-5fdb4cd5c7-5jj6p                   1/1     Running     0          2m26s
rook-ceph     rook-ceph-mon-b-549fb679cf-cfdlf                   1/1     Running     0          2m15s
rook-ceph     rook-ceph-mon-c-565ccf4667-4tbcz                   1/1     Running     0          2m5s
rook-ceph     rook-ceph-operator-57f48c4f45-dg625                1/1     Running     0          6m9s
rook-ceph     rook-ceph-osd-0-6c7f67586c-7lq9x                   1/1     Running     0          93s
rook-ceph     rook-ceph-osd-1-8664c8fc-ts57x                     1/1     Running     0          88s
rook-ceph     rook-ceph-osd-2-7f4b5ffddf-q44tx                   1/1     Running     0          87s
rook-ceph     rook-ceph-osd-prepare-set1-data-0-n67v9-qgf5w      0/1     Completed   0          108s
rook-ceph     rook-ceph-osd-prepare-set1-data-1-4kcdc-vql6d      0/1     Completed   0          108s
rook-ceph     rook-ceph-osd-prepare-set1-data-2-hcmc2-h24t8      0/1     Completed   0          107s
rook-ceph     rook-ceph-tools-7bb5797c8-5gdxx                    1/1     Running     0          6m8s
rook-ceph     rook-discover-8szlm                                1/1     Running     0          3m6s

$ kubectl get pvc -A
NAMESPACE   NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
default     datadir-cockroachdb-0   Bound    pvc-2613dbc9-658a-4d85-abc9-348c511e0dbb   16Gi       RWO            rook-ceph-block   44s
rook-ceph   set1-data-0-n67v9       Bound    local-blk1                                 32Gi       RWO            local-sc          2m14s
rook-ceph   set1-data-1-4kcdc       Bound    local-blk2                                 32Gi       RWO            local-sc          2m12s
rook-ceph   set1-data-2-hcmc2       Bound    local-blk3                                 32Gi       RWO            local-sc          2m10s
rook-ceph   set1-metadata-0-tr66s   Bound    local-mem1                                 3Gi        RWO            local-sc          2m13s
rook-ceph   set1-metadata-1-dbz4r   Bound    local-mem2                                 3Gi        RWO            local-sc          2m11s
rook-ceph   set1-metadata-2-dz97v   Bound    local-mem6                                 3Gi        RWO            local-sc          2m10s
rook-ceph   **set1-waldata-0-ncz55**    Bound    local-mem4                                 3Gi        RWO            local-sc          2m13s
rook-ceph   **set1-waldata-1-k5962**    Bound    local-mem3                                 3Gi        RWO            local-sc          2m11s
rook-ceph   **set1-waldata-2-md7tw**    Bound    local-mem5                                 3Gi        RWO            local-sc          2m9s